pkg/util/azureclient/mgmt/redhatopenshift/openshiftclusters_addons.go

```diff
@@ -5,15 +5,105 @@ package redhatopenshift
 
 import (
 	"context"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/validation"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 
 	"github.com/Azure/ARO-RP/pkg/client/services/preview/redhatopenshift/mgmt/2019-12-31-preview/redhatopenshift"
 )
 
+type OpenShiftClusterListOrGetKubernetesObjectsParameters struct {
+	Kind      string
+	Namespace string
+	Name      string
+}
+
 // OpenShiftClustersClientAddons contains addons for OpenShiftClustersClient
 type OpenShiftClustersClientAddons interface {
 	CreateOrUpdateAndWait(ctx context.Context, resourceGroupName string, resourceName string, parameters redhatopenshift.OpenShiftCluster) error
 	List(ctx context.Context) (clusters []redhatopenshift.OpenShiftCluster, err error)
 	ListByResourceGroup(ctx context.Context, resourceGroupName string) (clusters []redhatopenshift.OpenShiftCluster, err error)
+	ListOrGetKubernetesObjects(ctx context.Context, resourceGroupName string, resourceName string, parameters OpenShiftClusterListOrGetKubernetesObjectsParameters) (*unstructured.UnstructuredList, error)
+}
+
+func (c *openShiftClustersClient) ListOrGetKubernetesObjects(ctx context.Context, resourceGroupName string, resourceName string, parameters OpenShiftClusterListOrGetKubernetesObjectsParameters) (result *unstructured.UnstructuredList, err error) {
```

Comment:
I don't really want us to have to maintain all this, and I don't think it's the right place either.  In C# we'll be using some sort of dumb HTTP client call and treating the inputs and outputs opaquely - I think we should aim to do the same in Go too.